### ISSUE TITLE
Rule Success/Failure Callback

### ISSUE THREAD

**Issue by ehusaint** [CacheControl/json-rules-engine#16]
Currently I'm getting the complete rule object back as part of rule failurecallback but for rule successCallback, I only get the event object. Is it possible to get the full rule object back as part of the successCallback.

**Comment by CacheControl** [comment]
@ehusaint That's not currently a feature - the event params are meant to store all information necessary for event handling, however we can certainly discuss making this possible.   Can you elaborate a bit on why you're trying to access the rules object?  I'd like to better understand the use case.

**Comment by ehusaint** [comment]
I wanted to access the priority of the rule on successCallback but on successCallback, I only get back the event params and on failureCallback, I get back the complete rule params. On successCallback, I wanted to the know the priority of that rule. Is that possible?

**Comment by CacheControl** [comment]
@ehusaint Currently, you'd could do a little pseudo hack during runtime to attach the rule priority to the event, something like:

```js
let rule = new Rule()
rule.setConditions({ /* ... */ })
rule.setPriority(10)

rule.setEvent({
  type: 'message',
  params: {
    data: 'hello-world!',
    rulePriority: rule.priority // store the rule priority on the event
  }
})
```

I realize this isn't ideal, and I think I agree that the rule object should be more accessible.  I'll tag this as a feature request, however I'm about to go on holiday so it'll be a few weeks before I can provide a more elegant solution that what's above.

**Comment by ehusaint** [comment]
Got it. Yes, I can certainly use this as a workaround. But it would be certainly good to have that rule object more accessible as you mentioned. It would be great added feature. 

Thanks for a quick response.

**Comment by CacheControl** [comment]
This feature is now available in **2.0.0-beta1**.  There is now a 3rd argument provided to both the success and failure events which contain a clone of the rule object.  There's an [example](https://github.com/CacheControl/json-rules-engine/blob/7974907ba41a78a8ac6a195229ea04c28b8fd2de/examples/09-rule-results.js) in pr #35 and some docs [here](https://github.com/CacheControl/json-rules-engine/blob/7974907ba41a78a8ac6a195229ea04c28b8fd2de/docs/rules.md#rule-results) and [here](https://github.com/CacheControl/json-rules-engine/blob/7974907ba41a78a8ac6a195229ea04c28b8fd2de/docs/rules.md#events)

This a beta because I'm going to let it stew for a week or so to provide the various stakeholders an opportunity to give feedback.  Please let me know if you have any input.

**Comment by CacheControl** [comment]
Closing - access to rule properties now provided by the success/failure callback `ruleResult` argument
